Add plugin: Mahgen Renderer

MichaelFW-ui opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/MichaelFW-ui/obsidian-mahgen-plugin

## Release Checklist
- [ ] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[# Obsidian-mahgen-plugin](https://github.com/MichaelFW-ui/obsidian-mahgen-plugin/blob/dc858fd88f5eb70c31729604d9c45e8c16b7a2f6/README.md#L1)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.

[this.app.workspace.on('layout-change', () => {](https://github.com/MichaelFW-ui/obsidian-mahgen-plugin/blob/dc858fd88f5eb70c31729604d9c45e8c16b7a2f6/main.ts#L138)
Events need to be registered, so they can be properly unloaded when the plugin is disabled. <https://docs.obsidian.md/Plugins/Events>

[this.extension = markdownView?.getState().source ? [] : this.extension;](https://github.com/MichaelFW-ui/obsidian-mahgen-plugin/blob/dc858fd88f5eb70c31729604d9c45e8c16b7a2f6/main.ts#L140-L141)
Instead of this expensive operation check if the `livePreviewState` state field is true or not inside of your decoration in the `update` function.

MichaelFW-ui commented:
> [# Obsidian-mahgen-plugin](https://github.com/MichaelFW-ui/obsidian-mahgen-plugin/blob/dc858fd88f5eb70c31729604d9c45e8c16b7a2f6/README.md#L1) Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.
> 
> [this.app.workspace.on('layout-change', () => {](https://github.com/MichaelFW-ui/obsidian-mahgen-plugin/blob/dc858fd88f5eb70c31729604d9c45e8c16b7a2f6/main.ts#L138) Events need to be registered, so they can be properly unloaded when the plugin is disabled. https://docs.obsidian.md/Plugins/Events
> 
> [this.extension = markdownView?.getState().source ? [] : this.extension;](https://github.com/MichaelFW-ui/obsidian-mahgen-plugin/blob/dc858fd88f5eb70c31729604d9c45e8c16b7a2f6/main.ts#L140-L141) Instead of this expensive operation check if the `livePreviewState` state field is true or not inside of your decoration in the `update` function.

Thank you for the feedback. I’ve addressed the issues as follows:
1. Updated the plugin repo name. It is now https://github.com/MichaelFW-ui/mahgen-renderer .
2. Removed the expensive operation involving extension and replaced it with a check for livePreviewState in the update function.
3. Since the state check is handled properly, the event is no longer needed and has therefore been deleted.

Any further issues, please let me know. I am eager to learn from you. Thank you again.
